### PR TITLE
CC-4287 Resolve duplicate event plugins

### PR DESCRIPTION
**Pull request by m7moud**
- Developer(s): @m7moud

- Ticket: https://spryker.atlassian.net/browse/CC-4287

- Academy PR: -

- Release Group: https://release.spryker.com/release-groups/view/1485

#### Please confirm

- [X] No new OS components - or they have been approved by legal department

#### Documentation

- [X] Functional documentation provided or in progress?
- [X] Integration guide for projects provided or not needed?
- [X] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   EventBehavior         | minor                 |                       |

#### Release Notes

Ensure that events are only queued once.

-----------------------------------------

##### Change log

Improvements

- Adjusted `EventResourcePluginResolver` to remove duplicate plugins for the same event and resource.

### CODE REVIEW

File: src/Spryker/Zed/EventBehavior/Business/Model/EventResourcePluginResolver.php [hunk 1]
```diff
@@ -146,17 +146,21 @@ protected function getEffectivePlugins(array $resources): array
...
+                $pluginsPerExporter[static::REPOSITORY_EVENT_RESOURCE_PLUGINS][$effectivePlugin->getResourceName()][$effectivePlugin->getEventName()] = $effectivePlugin;
             }
+
             if ($effectivePlugin instanceof EventResourceQueryContainerPluginInterface) {
-                $pluginsPerExporter[static::QUERY_CONTAINER_EVENT_RESOURCE_PLUGINS][] = $effectivePlugin;
+                $pluginsPerExporter[static::QUERY_CONTAINER_EVENT_RESOURCE_PLUGINS][$effectivePlugin->getResourceName()][$effectivePlugin->getEventName()] = $effectivePlugin;
             }
         }
 
+        $pluginsPerExporter[static::REPOSITORY_EVENT_RESOURCE_PLUGINS] = array_map('current', $pluginsPerExporter[static::REPOSITORY_EVENT_RESOURCE_PLUGINS]);
```
Comment (ehsanmx) [inline src/Spryker/Zed/EventBehavior/Business/Model/EventResourcePluginResolver.php]: can you extract this to a separate method with a good name? like `resolveReposirotyPlugins()` or something like that

File: src/Spryker/Zed/EventBehavior/Business/Model/EventResourcePluginResolver.php [hunk 2]
```diff
@@ -146,17 +146,21 @@ protected function getEffectivePlugins(array $resources): array
...
+        $pluginsPerExporter[static::QUERY_CONTAINER_EVENT_RESOURCE_PLUGINS] = array_map('current', $pluginsPerExporter[static::QUERY_CONTAINER_EVENT_RESOURCE_PLUGINS]);
```
Comment (ehsanmx) [inline src/Spryker/Zed/EventBehavior/Business/Model/EventResourcePluginResolver.php]: same, ` array_map('current', ..)` deosn't tell why.

Comment (m7moud) [reply src/Spryker/Zed/EventBehavior/Business/Model/EventResourcePluginResolver.php]: `array_map('current', ..)` on a multi-dimensional array, will return a flattened array with only the only (the first) of any duplicates.

Comment (ehsanmx) [reply src/Spryker/Zed/EventBehavior/Business/Model/EventResourcePluginResolver.php]: `getFalttendReposirotyPluginsArray()` ;)

**Pull request merged**